orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/DelegatingClouddriverService.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver;
+
+import com.netflix.spinnaker.orca.ExecutionContext;
+import com.netflix.spinnaker.orca.clouddriver.config.SelectableService;
+
+class DelegatingClouddriverService<T> {
+  private final SelectableService selectableService;
+
+  DelegatingClouddriverService(SelectableService selectableService) {
+    this.selectableService = selectableService;
+  }
+
+  T getService() {
+    ExecutionContext executionContext = ExecutionContext.get();
```

Comment:
maybe worth handling the case of no threadlocal info here in case this gets called outside of a context where it is populated?